pmlc/conversion/gpu/gpu_to_vulkan.cc

```diff
@@ -265,11 +265,21 @@ ConvertGpuLaunchFuncToVulkanCalls::bindBuffers(Location loc, OpBuilder &builder,
           builder.getI32IntegerAttr(numElement * elementTypeSize));
       Value unrankedBuffer = builder.create<mlir::MemRefCastOp>(
           loc, buffer, getUnrankedMemRefType(elementType));
+
+      Value bufferType;
+      if (auto arg = buffer.dyn_cast<mlir::BlockArgument>()) {
+        bufferType = builder.create<LLVM::ConstantOp>(
```

Comment:
wondering if we should use something like `isBlockArgument` and set to `true` here instead of the 1/0 uint32_t type system you have now.  Or, `isHostMemory` ?

Reply:
Yes, that is better. Using isBlockArgument and bool now.